mtp_api/apps/prison/views.py

```diff
@@ -32,3 +34,29 @@ def create(self, request, *args, **kwargs):
 
     def perform_create(self, serializer):
         serializer.save(created_by=self.request.user)
+
+
+class PrisonerValidityView(mixins.ListModelMixin, viewsets.GenericViewSet):
+    queryset = PrisonerLocation.objects.all()
+    permission_classes = (
+        IsAuthenticated, SendMoneyClientIDPermissions,
+    )
+    serializer_class = PrisonerValiditySerializer
+
+    def filter_queryset(self, queryset):
+        queryset = super().filter_queryset(queryset)
+        return queryset.filter(
+            prisoner_number=self.request.GET['prisoner_number'],
+            prisoner_dob=self.request.GET['prisoner_dob'],
+        )
+
+    def list(self, request, *args, **kwargs):
+        prisoner_number = self.request.GET.get('prisoner_number', '')
+        prisoner_dob = self.request.GET.get('prisoner_dob', '')
+        try:
+            prisoner_dob = parse_date(prisoner_dob)
+        except ValueError:
+            prisoner_dob = None
+        if not prisoner_number or not prisoner_dob:
+            return Response(None, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
I think we should return an error message here. Just something saying 'prisoner_dob is required' and 'prisoner_number is required', in the same format as that used for the conflict errors in the cashbook/bank-admin APIs.